File: 704-binary-search/704-binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */


var search = function(nums, target) {
    let high = nums.length-1;
    let low = 0; 
    
    while (low <=high){
        let med = Math.floor((high+low)/2)
        if(nums[med]>target){
            high = med -1;
        }
        else if (nums[med]<target){
            low = med+1;
        }
        else {
            return med;
        }
    }
    return -1
};

File: 704-binary-search/704-binary-search.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer}
def search(nums, target)
    high = nums.length-1
    low = 0
    while low<= high do
        med = (high+low)/2
        if nums[med] > target
            high = med-1;
        elsif nums[med] < target
            low= med+1;
        else 
            return med;
        end
    end
    return -1;
end